pyproject.toml

```diff
@@ -6,3 +6,77 @@ requires = [
 backend-path = [
   "_custom_build",
 ]
+
+[project]
+name = "pillow"
+description = "Python Imaging Library (Fork)"
+readme = "README.md"
+keywords = [
+  "Imaging",
+]
+license = {text = "HPND"}
+authors = [{name = "Jeffrey A. Clark (Alex)", email = "aclark@aclark.net"}]
+requires-python = ">=3.8"
+classifiers = [
+  "Development Status :: 6 - Mature",
+  "License :: OSI Approved :: Historical Permission Notice and Disclaimer (HPND)",
+  "Programming Language :: Python :: 3 :: Only",
+  "Programming Language :: Python :: 3.8",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: 3.12",
+  "Programming Language :: Python :: Implementation :: CPython",
+  "Programming Language :: Python :: Implementation :: PyPy",
+  "Topic :: Multimedia :: Graphics",
+  "Topic :: Multimedia :: Graphics :: Capture :: Digital Camera",
+  "Topic :: Multimedia :: Graphics :: Capture :: Screen Capture",
+  "Topic :: Multimedia :: Graphics :: Graphics Conversion",
+  "Topic :: Multimedia :: Graphics :: Viewers",
+]
+dynamic = [
+  "version",
+]
```

Comment:
Looking at https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html#dynamic-metadata,
should this not be
```
dynamic = ["version", "readme"]
```
and move `readme = "README.md"` from earlier into
```
[tool.setuptools.dynamic]
version = {attr = "PIL.__version__"}
readme = {file = "README.md"}
```
later?

Reply:
The `readme` is listed in both examples at that page, but it looks to me like it should be in the dynamic section so that the content type can be specified (the default is `text/x-rst`):
```
readme = {file = "README.md", content-type = "text/markdown"}
```

Reply:
Version:

Setting the version is good, it means we can move `version=PILLOW_VERSION` out of `setup.py`'s `setup()`.

README:

Despite the docs saying content-type defaults to `text/x-rst`, I did a test upload to https://test.pypi.org/project/pillow/10.2.0.dev0/ (files built using `python -m build` from the current state of the PR) and the README renders just fine, so I think it's fine?

PS @radarhere and other Pillow maintainers: if you like, I can add you to https://test.pypi.org/project/pillow/ in case you want to test anything out there. Just let me know your username (you need to create a new account there).